Source/SwiftLintCore/Extensions/Configuration+RulesMode.swift

```diff
@@ -101,6 +104,9 @@ public extension Configuration {
             case let .only(onlyRules):
                 return .only(Set(onlyRules.map(aliasResolver)))
 
+            case let .onlyRule(onlyRules):
+                return .onlyRule(Set(onlyRules.map(aliasResolver)))
```

Comment:
So that would be nicer, and the original code tried to implement `--only-rule` via `.only`. The problem is that when we're merging any later configurations, generally we want `--only-rule` to override them, as it's been specified on the command line, whereas with `only_rules`, the child configuration should take precedence.

Totally agree about the multiple identifiers - I do have a follow-on PR that allows `--only-rule` to be specified multiple times on the command line, to enable multiple rules, so it's slightly more understandable in that context.

It would be better if the naming were clearer - both `allEnabled` and `onlyRule` are really command line options, and it would be nice if their naming captured that somehow, or at least if `onlyRule` was somehow more distinctive, but I don't have any elegant suggestions.

Reply:
I see. Perhaps we could be more specific with the mode names having for example `.defaultConfiguration`, `.onlyConfiguration`, `.onlyCommandLine` and `.allCommandLine`.

Allowing multiple `--only-rule` command line arguments makes it more reasonable that `.onlyRule` takes a set of IDs. Let's document that at its declaration as well.